### PR TITLE
[CHK-2388] - part 3 feat: receiver stop business logic implementation

### PR DESCRIPTION
**Pull request by pietro-tota**
Depends on #198 

#### List of Changes
Add logic to query and start/stop event receivers using SpEL Control Bus integration

#### Motivation and Context
Add implementation to handle and query receivers status

#### How Has This Been Tested?
Local tests + junit tests + test in DEV environment

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by Fernando-Granato — APPROVED** [review]

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/EventReceiverService.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    if (lastStatuses.isEmpty()) {
+      throw NoEventReceiverStatusFound()
+    }
+    return EventReceiverStatusResponseDto(status = lastStatuses)
+  }
+
+  @PreDestroy
+  fun close() {
+    RedisStreamMessageSource.RedisStreamMessageSourceLogger.logger.info(
+      "Deleting consumer group with id $redisStreamConf.consumerGroup")
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/EventReceiverService.kt:72]:
```suggestion
      "Deleting consumer group with id ${redisStreamConf.consumerGroup}")
```

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/EventReceiverStatusPoller.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  @Autowired
+  private val eventDispatcherReceiverStatusTemplateWrapper:
+    EventDispatcherReceiverStatusTemplateWrapper,
+  @Autowired private val inboundChannelAdapterHandlerService: InboundChannelAdapterHandlerService,
+  @Autowired private val redisStreamEventControllerConfigs: RedisStreamEventControllerConfigs
+) {
+
+  private val logger = LoggerFactory.getLogger(javaClass)
+
+  @Scheduled(cron = "\${eventController.status.pollingChron}")
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/EventReceiverStatusPoller.kt:28]: Polling defeats a bit the purpose of Pub/Sub, we should consider refactoring this to use Spring Data Redis' [`StreamReceiver`](https://docs.spring.io/spring-data/redis/reference/redis/redis-streams.html#reactive-streamreceiver) later on

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/InboundChannelAdapterHandlerService.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import org.springframework.integration.annotation.InboundChannelAdapter
+import org.springframework.integration.channel.DirectChannel
+import org.springframework.integration.channel.QueueChannel
+import org.springframework.messaging.MessageHeaders
+import org.springframework.messaging.support.MessageBuilder
+import org.springframework.stereotype.Service
+
+/** This class handles InboundChannelAdapter lifecycle through SpEL ControlBus implementation */
+@Service
+class InboundChannelAdapterHandlerService(
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/InboundChannelAdapterHandlerService.kt:18]: 🚲 
```suggestion
class InboundChannelAdapterLifecycleHandlerService(
```

**Review by giovanniberti — APPROVED** [review]

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/EventReceiverStatusPoller.kt (hunk 2 above)
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/EventReceiverStatusPoller.kt:30]: This polling class is used to update each event dispatcher instance event receivers statuses

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/InboundChannelAdapterHandlerService.kt (hunk 3 above)
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/InboundChannelAdapterHandlerService.kt]: Fixed with https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/199/commits/894d1be5512a584bffecbec6ea2b4533f1f988b6

**Pull request merged**